### PR TITLE
Pass addresses and map name of userspace instrumentation to LinuxTracing

### PR DESCRIPTION
**Pull request by dpallotti**
We are passing the address ranges of user space instrumentation trampolines to
`LinuxTracing`. We are also passing the map name of the injected library. We
need these for the following purposes:

- Detecting if a sample fell directly in a trampoline (entry or return), in
  `UprobesUnwindingVisitor`;
- Detecting if a sample fell in the injected library, in
  `UprobesUnwindingVisitor`;
- Detecting if unwinding stopped at a return trampoline, in
  `UprobesUnwindingVisitor`;
- Patching frame pointer callchains in `UprobesReturnAddressManager`.

We hold the address ranges and the injected library's map name with the
`UserSpaceInstrumentationAddresses` interface.

Actual usage of this information will come as the next change.

Note that with uprobes the approach was simpler, as most of the work for uprobes
happens in the kernel, for which we don't get samples, and for the rest we could
simply use the map name "[uprobes]", corresponding to a couple of addresses for
the return trampoline.

Bug: http://b/194704608
Bug: http://b/204404077

Test: Unit tests. Tried as part of the integration of user space instrumentation
and unwinding as per
http://go/stadia-orbit-user-space-instrumentation-and-unwinding#heading=h.a3i5cfh8pexm,
also with frame pointers on custom target.

### CODE REVIEW

**Comment by dpallotti** [comment]
For a preview of the actual usage in `LinuxTracing`, I created draft https://github.com/google/orbit/pull/2985.

File: src/LinuxTracing/include/LinuxTracing/UserSpaceInstrumentationAddresses.h [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+#ifndef LINUX_TRACING_USER_SPACE_INSTRUMENTATION_ADDRESSES_H_
+#define LINUX_TRACING_USER_SPACE_INSTRUMENTATION_ADDRESSES_H_
+
+namespace orbit_linux_tracing {
+
+// This interface carries information about memory addresses dedicated to user space
+// instrumentation, in particular the entry trampolines, the return trampoline, and the injected
+// library. It allows querying whether an address belongs to a trampoline, and the map name of the
+// injected library.
+class UserSpaceInstrumentationAddresses {
```
Comment (dpallotti) [inline src/LinuxTracing/include/LinuxTracing/UserSpaceInstrumentationAddresses.h:14]: If you have a suggestion for a better name, please come forward!

**Review by danielfenner — APPROVED** [review]
lgtm, thanks!

File: src/LinuxTracing/include/LinuxTracing/UserSpaceInstrumentationAddresses.h (hunk 1 above)
Comment (florian-kuebler) [reply src/LinuxTracing/include/LinuxTracing/UserSpaceInstrumentationAddresses.h:14]: I would probably call it `UserSpaceInstrumentationAddressRanges`, but find the other suggestions also fine.

**Review by florian-kuebler — APPROVED** [review]
LGTM

File: src/LinuxTracing/include/LinuxTracing/UserSpaceInstrumentationAddresses.h (hunk 1 above)
Comment (dpallotti) [reply src/LinuxTracing/include/LinuxTracing/UserSpaceInstrumentationAddresses.h:14]: I feel kind of equally about all of them so I guess I'll just keep it as it is in the interest of choosing the shortest name. Thanks for the suggestions!

**Pull request merged**